include/nbl/config/BuildConfigOptions.h.in

```diff
@@ -70,4 +70,16 @@
 // TODO: This has to disapppear from the main header and go to the OptiX extension header + config
 #cmakedefine OPTIX_INCLUDE_DIR "@OPTIX_INCLUDE_DIR@"
 
-#endif
\ No newline at end of file
+// DDL exports
+#cmakedefine _NBL_SHARED_BUILD_
+#ifdef _NBL_SHARED_BUILD_
+#if defined(_NBL_PLATFORM_WINDOWS_) && defined(_MSC_VER)
+#define NBL_EXPORTS __declspec(dllexport)
+#else
+#pragma error("Shared builds are only available for Windows OS & MSVC target for now!")
+#endif
+#else
+#define NBL_EXPORTS
+#endif
```

Comment:
got it backwards, follow my recipe, rename `NBL_EXPORTS` `NBL_API`

NBL_API needs to be `dllexport` when you build nabla, but `dllimport` when you use it

_NBL_SHARED_BUILD_ is nice, it tells me whether I have a static or shared build

```c
#cmakedefine _NBL_SHARED_BUILD_
#ifdef _NBL_SHARED_BUILD_
    #if defined(__NBL_BUILDING_NABLA__) // currently compiling nabla, this define is passed through the commandline
        #if defined(_MSC_VER)
            #define NBL_API __declspec(dllexport)
        #elif defined(__GNUC__)
            #define NBL_API __attribute__ ((visibility ("default")))
        #endif
    #else
        #if defined(_MSC_VER)
            #define NBL_API __declspec(dllimport)
        #else
            #define NBL_API
        #endif
    #endif
#else
    #define NBL_API
#endif
   ```

Reply:
done